app/src/main/java/com/dankideacentral/dic/LaunchActivity.java

```diff
@@ -0,0 +1,39 @@
+package com.dankideacentral.dic;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.preference.PreferenceManager;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+
+public class LaunchActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String twitterAuth = preferences.getString(getString(R.string.twitter_auth_preference), "");
+        String fineLocationPermission = "android.permission.ACCESS_FINE_LOCATION";
+        PackageManager pm = this.getPackageManager();
+        int hasFineLocationPermission = pm.checkPermission(
+                fineLocationPermission,
+                this.getPackageName());
+        // auth tokens don't expire. Just check if it exists
+        if ("".equals(twitterAuth)) {
```

Comment:
So create an "empty string" Constant? I was only doing it this way so I didn't have to null check twitterAuth.